### PR TITLE
feat: make vite the default `create-fuels` template

### LINKED ISSUE #3004: Make `vite` the default template for `create fuels`
Before we can make `vite` the default template for `create fuels` we need to do the following:

- Adjust all onboarding documentation including the quick start to reference the `vite` template
- Adjust `create fuels` tests to support the new default template, including the integration test that only runs in CI
- Document template selection in the `create fuels` documentation

### PR DESCRIPTION
**Pull request by Dhaiwat10**
- Closes #3004 

# Release notes

In this release, we:

- Made `vite` the default template for `create-fuels`

# Summary

This PR adjusts our docs and the CLI code for `create-fuels` to make `vite` the new default template for the tool.

# Breaking Changes

If no `--template` is specified, `npm create fuels` will now bootstrap a Vite app instead of a Next.js app.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -69,23 +69,23 @@ jobs:
       - name: Run UI tests (on template)
         run: sh ./scripts/tests-ui.sh
```
Comment (petertonysmith94) [inline .github/workflows/test.yaml:70]: I wonder if when we spin up `fuel-core` here, it's interacting with the other `fuel-core` from the published step?

Comment (petertonysmith94) [reply .github/workflows/test.yaml:70]: May be worth disabling this and seeing if it passes

Comment (danielbate) [reply .github/workflows/test.yaml]: It's a nice idea, will try

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -80,12 +80,12 @@ jobs:
       - name: Run tests (with published version)
         run: pnpm test:integration
         env:
-          PUBLISHED_NPM_TAG: next
+          PUBLISHED_NPM_TAG: pr-3048
```
Comment (danielbate) [inline .github/workflows/test.yaml:83]:
```suggestion
          PUBLISHED_NPM_TAG: next
```

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -80,12 +80,12 @@ jobs:
...
 
       - name: Run UI tests (with published version)
         run: sh ./scripts/tests-ui-integration.sh
         env:
-          PUBLISHED_NPM_TAG: next
+          PUBLISHED_NPM_TAG: pr-3048
```
Comment (danielbate) [inline .github/workflows/test.yaml:88]:
```suggestion
          PUBLISHED_NPM_TAG: next
```

**Review by danielbate — DISMISSED** [review]
Great work 💪🏻

File: playwright-tests/create-fuels.test.ts [hunk 4]
```diff
@@ -11,9 +11,13 @@ test.extend({
 });
 
 test('counter contract - increment function call works properly', async ({ page }) => {
+  // make a get request to localhost:4000/v1/playground to check if the node is running
+  const response = await fetch('http://localhost:4000/v1/playground');
```
Comment (petertonysmith94) [inline playwright-tests/create-fuels.test.ts:15]: A `waitUntilReachable` method might be useful

File: playwright-tests/create-fuels.test.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { test, expect } from '@playwright/test';
 
-const WEB_SERVER_URL = 'http://127.0.0.1:3000';
+const WEB_SERVER_URL = 'http://localhost:5173';
```
Comment (petertonysmith94) [inline playwright-tests/create-fuels.test.ts:3]: Why did we change the hostname?

**Review by petertonysmith94 — APPROVED** [review]
Nice 🚀

File: playwright-tests/create-fuels.test.ts (hunk 4 above)
Comment (Dhaiwat10) [reply playwright-tests/create-fuels.test.ts:15]: Oh we don't need this code anyways, this was just for debugging 😅  Removing it

File: playwright-tests/create-fuels.test.ts (hunk 5 above)
Comment (petertonysmith94) [reply playwright-tests/create-fuels.test.ts:3]: Intersting.

**Review by danielbate — APPROVED** [review]

**Review by maschad — APPROVED** [review]

**Pull request merged**